Core/Src/ADXL.c

```diff
@@ -10,21 +10,86 @@
 #include "main.h"
 #include <stdbool.h>
 #include <errno.h>
-#include <stm32l4xx_hal_dma.h>
 
 extern SPI_HandleTypeDef hspi1;
-#define ADXL_SPI_ReadOp (1 << 7)
-#define ADXL_SPI_MB     (1 <<6)
-#define ADXL_INT_ENABLE_REG 0x2E
-#define ADXL_INT_MAP_REG    0x2F
-#define ADXL_DEVICEID       0x00
-bool ACC_EXTI_Interrupt;
-static uint8_t TxBuffer[] = { 0, 0, 0, 0, 0, 0, 0 };
-static uint8_t RxBuffer[7];
-
 extern DMA_HandleTypeDef hdma_spi1_rx;
 
-bool ADXL_WriteData(uint8_t regAddress, uint8_t registerValue) {
+#define ADXL_SPI_ReadOp             (1 << 7)
+#define ADXL_SPI_MB                 (1 <<6)
+#define ADXL_INT_ENABLE_REG         0x2E
+#define ADXL_INT_MAP_REG            0x2F
+#define ADXL_DEVICEID               0x00
+#define ADXL_DATAX0                 0x32
+#define ADXL_POWER_CTL              0x2D
+#define ADXL345_SCALE_FACTOR        39  //  if is 2g  , if is 4g *2  8g *4 16g*8
+
+static bool _ReadData(uint8_t regAddress, uint8_t *registerValue);
```

Comment:
Since static functions are only present in the .c file, you should provide a very brief description using the same standard doxygen documentation.